src/fmu/dataio/providers/objectdata/_faultroom.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Final
+
+from fmu.dataio._definitions import ValidFormats
+from fmu.dataio._logging import null_logger
+from fmu.dataio.datastructure.meta import meta, specification
+from fmu.dataio.readers import FaultRoomSurface
+
+from ._base import (
+    DerivedObjectDescriptor,
+    ObjectDataProvider,
+)
+
+logger: Final = null_logger(__name__)
+
+
+@dataclass
+class FaultRoomSurfaceProvider(ObjectDataProvider):
+    obj: FaultRoomSurface
+
+    def get_bbox(self) -> dict[str, Any]:
+        """Derive data.bbox for FaultRoomSurface."""
+        logger.info("Get bbox for FaultRoomSurface")
+        assert isinstance(self.obj, FaultRoomSurface)  # for mypy
```

Comment:
I think the `assert` should not be necessary with `obj: FaultRoomSurface` annotated in the class